src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/JourneyPriceCatalogueController.kt

```diff
@@ -27,30 +24,32 @@ private val logger = loggerFor<JourneyPriceCatalogueController>()
  * Controller responsible for generating and returning an on demand journey price catalogue spreadsheet to the end user.
  */
 @RestController
-@SessionAttributes(SUPPLIER_ATTRIBUTE)
 class JourneyPriceCatalogueController(
   private val journeyPriceCatalogueService: JourneyPriceCatalogueService,
   private val timeSource: TimeSource
 ) {
 
-  @GetMapping("/generate-prices-spreadsheet/{supplier}")
+  @GetMapping(GENERATE_PRICES_SPREADSHEET)
   @Throws(IOException::class)
   fun generateJourneyPriceCatalogue(
-    @PathVariable supplier: String,
-    @RequestParam(
-      name = "moves_from",
-      required = true
-    ) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) movesFrom: LocalDate,
-    response: HttpServletResponse?,
-    authentication: Authentication?
+    @SessionAttribute(name = SUPPLIER_ATTRIBUTE, required = false) supplier: Supplier?,
+    @SessionAttribute(name = DATE_ATTRIBUTE, required = false) movesFrom: LocalDate?,
+    response: HttpServletResponse?
   ): ResponseEntity<InputStreamResource?>? {
+
+    if (supplier == null || movesFrom == null) {
```

Comment:
If these are not present then there is nothing can do as we cannot serve a file.